src/org/jitsi/impl/neomedia/MediaStreamImpl.java

```diff
@@ -3635,6 +3636,89 @@ public void injectPacket(RawPacket pkt, boolean data, TransformEngine after)
     }
 
     /**
+    * {@inheritDoc}
+    */
+    public boolean isKeyFrame(byte[] buf, int off, int len)
+    {
+        if (buf == null || buf.length < off + len)
+        {
+            return false;
+        }
+
+        Byte redPT = null, vp8PT = null, h264PT = null;
+        for (Map.Entry<Byte, MediaFormat> entry :
+            getDynamicRTPPayloadTypes().entrySet())
+        {
+            String encoding = entry.getValue().getEncoding();
+            Byte payloadType = entry.getKey();
+
+            if (Constants.RED.equalsIgnoreCase(encoding))
+            {
+                redPT = payloadType;
+            }
+            else if (Constants.VP8.equalsIgnoreCase(encoding))
+            {
+                vp8PT = payloadType;
+            }
+            else if (Constants.H264.equalsIgnoreCase(encoding))
+            {
+                h264PT = payloadType;
+            }
+        }
+
+        int payloadOff, payloadLen;
+        byte payloadType;
+
+        try {
+          // XXX this will not work correctly when RTX gets enabled!
+            if (redPT != null
+                && redPT == RawPacket.getPayloadType(buf, off, len))
+            {
+                REDBlock block = REDBlockIterator
+                    .getPrimaryBlock(buf,
+                        RawPacket.getPayloadOffset(buf, off, len),
+                        RawPacket.getPayloadLength(buf, off, len));
```

Comment:
Thank you for adding the comment but I think you forgot to adapt the code accordingly.